src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -327,8 +355,10 @@ where
 
 impl<I: HummockIterator, NE: NodeExtraOrderInfo> Drop for MergeIteratorInner<I, NE> {
     fn drop(&mut self) {
-        let mut stats = StoreLocalStatistic::default();
-        self.collect_local_statistic_impl(&mut stats);
-        stats.report(self.stats.as_ref());
+        if let Some(stats) = &self.stats {
```

Comment:
Why do we need to report metrics when `MergeIterator` drops?

Reply:
`MergeIterator` is the top level iterator for compactors.

Reply:
I think the interface is still not friendly. I think refactor is needed after the fix.